### PR TITLE
update CMakeLists.txt file, so that it can be installed pro

### PR DESCRIPTION
**Pull request by Chunting**
Update CMakeLists.txt file, so that it can generate .cmake file properly.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -18,8 +18,9 @@
 cmake_minimum_required(VERSION 2.8)
 
 include(cmake/base.cmake)
-include(cmake/eigen.cmake)
 include(cmake/boost.cmake)
+include(cmake/cpack.cmake)
```
Comment (gergondet) [inline CMakeLists.txt]: I'm pretty sure this is not needed. See comments in `cpack.cmake` for better altenratives.

File: CMakeLists.txt [hunk 2]
```diff
@@ -28,11 +29,13 @@ set(PROJECT_URL "https://github.com/jrl-umi3218/mc_rbdyn_urdf")
 set(PROJECT_DEBUG_POSTFIX "_d")
 
 set(DOXYGEN_USE_MATHJAX "YES")
+SET(CXX_DISABLE_WERROR True)
 setup_project()
 
 option(PYTHON_BINDING "Generate python binding." ON)
 option(PYTHON_BINDING_USER_INSTALL "Install the Python bindings in user space" OFF)
 option(DISABLE_TESTS "Disable unit tests." OFF)
+option(BENCHMARKS "Generate benchmarks." OFF)
```
Comment (gergondet) [inline CMakeLists.txt]: Also not needed since there is not benchmarks to compile in the project.

File: CMakeLists.txt [hunk 3]
```diff
@@ -63,3 +66,7 @@ endif()
 pkg_config_append_libs(mc_rbdyn_urdf)
 
 setup_project_finalize()
+SETUP_PROJECT_CPACK()
```
Comment (gergondet) [inline CMakeLists.txt]: As noted in the documentation of the macro, this is kind-of outdated and better solution are built-in for archive generations.

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -21,6 +21,7 @@ set(SOURCES
 set(HEADERS
   ../include/mc_rbdyn_urdf/api.h
   ../include/mc_rbdyn_urdf/urdf.h
+  ../include/mc_rbdyn_urdf/kukaiiwaURDF.h
```
Comment (gergondet) [inline src/CMakeLists.txt]: Shouldn't be there.

File: src/CMakeLists.txt [hunk 5]
```diff
@@ -33,7 +34,10 @@ target_compile_definitions(mc_rbdyn_urdf PRIVATE "-DMCRBDYNURDF_BUILDING")
...
 install(FILES ${HEADERS} DESTINATION ${INSTALL_PATH})
 install(TARGETS mc_rbdyn_urdf
-  RUNTIME DESTINATION bin
-  LIBRARY DESTINATION lib
-  ARCHIVE DESTINATION lib
+EXPORT "${TARGETS_EXPORT_NAME}"
+LIBRARY DESTINATION "lib"
+ARCHIVE DESTINATION "lib"
+RUNTIME DESTINATION "bin"
+INCLUDES DESTINATION "${INCLUDE_INSTALL_DIR}"
```
Comment (gergondet) [inline src/CMakeLists.txt]: Please keep the indentation that was in the original file

File: src/CMakeLists.txt [hunk 6]
```diff
@@ -33,7 +34,10 @@ target_compile_definitions(mc_rbdyn_urdf PRIVATE "-DMCRBDYNURDF_BUILDING")
...
 )
+install(FILES ${HEADERS} DESTINATION ${INCLUDE_INSTALL_DESTINATION})
```
Comment (gergondet) [inline src/CMakeLists.txt]: This is redundant with `install(FILES ${HEADERS} DESTINATION ${INSTALL_PATH})`

**Review by gergondet — CHANGES_REQUESTED** [review]
Thanks for the PR. I'll merge it once the issues I've raised have been fixed.

It is also missing an update to the cmake submodule.

File: CMakeLists.txt (hunk 1 above)
Comment (Chunting) [reply CMakeLists.txt]: This line has been removed.

File: CMakeLists.txt (hunk 2 above)
Comment (Chunting) [reply CMakeLists.txt]: This line has been removed.

File: CMakeLists.txt (hunk 3 above)
Comment (Chunting) [reply CMakeLists.txt]: Also this line has been removed.

File: src/CMakeLists.txt (hunk 5 above)
Comment (Chunting) [reply src/CMakeLists.txt]: Is the indentation style four spaces or two spaces?  In [RBDy/src/CMakelist.txt](https://github.com/jrl-umi3218/RBDyn/blob/master/src/CMakeLists.txt), it's four spaces but in this file it's two spaces.

I changed it back to two spaces.

File: src/CMakeLists.txt (hunk 4 above)
Comment (Chunting) [reply src/CMakeLists.txt]: It has been removed and included in my project.

File: src/CMakeLists.txt (hunk 6 above)
Comment (Chunting) [reply src/CMakeLists.txt]: This line has also been removed.

**Pull request closed without merging**